packages/ui-components/src/NavButton.tsx

```diff
@@ -20,25 +20,53 @@ interface NavButtonProps extends StyledNavButtonProps {
 
 // TEMPORARY, to be moved to shared/constants
 const tachyons = {
-  default: 'bg-black white f4',
-  negative: 'bg-white black f4',
+  default: 'bg-black white f4 fw4',
+  negative: 'bg-white black f4 fw4',
+};
+const mergeClasses = (defaultClass: string, inClassName: string): string => {
```

Comment:
Hey @amaurymartiny I've stumbled on an annoying issue. Some (to most of the) components have a default set of tachyons classes which we want to keep rewritable. There are two simple ways to do it:
1. add two sets of classes as two strings => if in tachyons.css file an incoming class is declared before the default class, this won't affect appearance of the component
2. if there are incoming classes use only them => not nice

The third way is regex merge of incoming onto default classes like here. The function can be abstracted. Do you have any thoughts on this?

Reply:
I'm not sure I fully understand, but here's what I understood:

Let's take NavButton, in the DOM it's `<div className="{styled-components-stuff} bg-black white f4 fw4">`.

The best way would be, if user decides to overwrite some tachyons class, that we would just do:
```jsx
<NavButton className="bg-red" />
```

And it would keep all the other tachyons classes, but add a `bg-red`.

I think instead of the regex way you did, a simple string concatenation would be enough: `mergeClasss = defaultClass.concat(' ').concat(inClassName);` In this case, the red NavButton would have this DOM:
```jsx
<div className="{styled-components-stuff} bg-black white f4 fw4 bg-red">
```
I just tried on storybook, only the last bg is taken into account, I think subsequent tachyon classes just overwrite previous ones, which is exactly what we want.

Reply:
Also, instead of a `.concat()` on mergeClass, we could use something like https://github.com/JedWatson/classnames (I used it before, it's quite nice):
`<MyComponent className={classNames(defaultClass, inClassName)} />`

Reply:
> simple string concatenation would be enough

it is not always the case:

- `bg-red` rewrites `bg-black` just by being concatenated to the `bg-black`, but
- `bg-black` does not rewrite `bg-red`

this is why we need to remove `bg-red` when user adds `bg-black`. The same is true for ig `f7` and `f1`